app/components/UI/Stake/Views/UnstakeInputView/UnstakeInputView.test.tsx

```diff
@@ -161,4 +174,46 @@ describe('UnstakeInputView', () => {
       expect(screen.queryAllByText('Not enough ETH')).toHaveLength(2);
     });
   });
+
+  describe('when staking_transactions feature flag is enabled', () => {
+    let originalMock: jest.Mock;
+    let mockAttemptUnstakeTransaction: jest.Mock;
+
+    beforeEach(() => {
+      originalMock = jest.requireMock('../../../../../selectors/featureFlagController').selectConfirmationRedesignFlags as jest.Mock;
+
+      jest.requireMock('../../../../../selectors/featureFlagController').selectConfirmationRedesignFlags = jest.fn(() => ({
+        staking_transactions: true,
+      }));
+
+      mockAttemptUnstakeTransaction = jest.fn().mockResolvedValue(undefined);
+      jest.requireMock('../../hooks/usePoolStakedUnstake').default = () => ({
+        attemptUnstakeTransaction: mockAttemptUnstakeTransaction,
+      });
+    });
+
+    afterEach(() => {
+      jest.requireMock('../../../../../selectors/featureFlagController').selectConfirmationRedesignFlags = originalMock;
+    });
+
+    it('calls attemptUnstakeTransaction when Review button is pressed', async () => {
+      render(UnstakeInputView);
+
+      fireEvent.press(screen.getByText('1'));
+
+      fireEvent.press(screen.getByText('Review'));
+
+      // Wait for the async operation to complete
+      await new Promise(resolve => setTimeout(resolve, 0));
+
+      expect(mockAttemptUnstakeTransaction).toHaveBeenCalled();
+      expect(mockNavigate).toHaveBeenCalledWith('StakeScreens', {
+        screen: Routes.STANDALONE_CONFIRMATIONS.STAKE_WITHDRAWAL,
+        params: expect.objectContaining({
+        amountWei: expect.any(String),
+          amountFiat: expect.any(String),
+        }),
+      });
```

Comment:
`waitFor` should work if you're waiting for rendering updates before making assertions.

Alternatively, you can use `await act(async () => {...})` to wait for rendering update for individual actions.

```suggestion
      fireEvent.press(screen.getByText('1'));

      fireEvent.press(screen.getByText('Review'));

      // Wait for the async operation to complete
     await waitFor(async () => {
      expect(mockAttemptUnstakeTransaction).toHaveBeenCalled();
      expect(mockNavigate).toHaveBeenCalledWith('StakeScreens', {
        screen: Routes.STANDALONE_CONFIRMATIONS.STAKE_WITHDRAWAL,
        params: expect.objectContaining({
        amountWei: expect.any(String),
          amountFiat: expect.any(String),
        }),
      });
     });
    
```

Reply:
I tried `await waitFor` when implementing the test and it didn't pass. I see that using `flushPromises` is a common pattern in the repo, so I think it should be fine using the set timeout promise to flush promises here as well.

Reply:
Sounds good!